homeworks/ovarikristof/3_conditionals_loops/rock_paper_scissors.py

```diff
@@ -0,0 +1,85 @@
+ BEKÉRÜNK NEVEKET A FELHASZNÁLÓKHOZ:
+player_1 = input("Please enter a name for Player 1: ")
+player_2 = input("Please enter a name for Player 2: ")
+
+# BEKÉRJÜK, HOGY HÁNY KÖRT AKARNAK JÁTSZANI,
+#ÉS MEGHATÁROZZUK, HOGY CSAK PÁRATLAN KÖRÖK MEGADÁSA UTÁN INDUL A JÁTÉK
+def game_rounds():
+    while True:
+        try:
+            rounds_want_to_play = int(input("Please enter how many rounds do you want to play: "))
+            if rounds_want_to_play % 2 == 1:
+                return rounds_want_to_play
+            else:
+                print("Please enter an odd number, else there won't be a winner!")
+        except ValueError:
+            print("Please enter a valid number!")
+
+
+#MEGKÉRJÜK A JÁTÉKOSOKAT, HOGY ADJÁK MEG A VÁLASZAIKAT
+def get_players_1_choice(player_1):
+    while True:
+        player_1_choice = input(f"{player_1}, please enter your choice (rock, paper, scissors): ").lower()
+        if player_1_choice in ["rock", "paper", "scissors"]:
+            return player_1_choice
+        else:
+            print("Please enter only the following: rock, paper, scissors!")
+
+def get_players_2_choice(player_2):    
+    while True:
+        player_2_choice = input(f"{player_2}, please enter your choice (rock, paper, scissors): ").lower()
+        if player_2_choice in ["rock", "paper", "scissors"]:
+            return player_2_choice
+        else:
+            print("Please enter only the following: rock, paper, scissors!")
+
+#MEGHATÁROZZUK, HOGY A JÁTÉKOSOK VÁLASZAI ALAPJÁN MELYIKÜK AZ AMELYIK PONTOT SZEREZ A VÁLASZTÁSA ALAPJÁN
+def determine_winner(choice_1, choice_2):
+    if choice_1 == choice_2:
+        return 0                                                 #DÖNTETLEN
+    elif (choice_1 == "rock" and choice_2 == "scissors") or \
+         (choice_1 == "scissors" and choice_2 == "paper") or \
+         (choice_1 == "paper" and choice_2 == "rock"):
+        return 1                                                 #PLAYER_1 NYERT
+    else:
+        return 2                                                 #PLAYER_2 NYERT
+
+#A JÁTÉK DEFINIÁLÁSA:
+def rock_paper_scissors():
+    rounds_want_to_play = game_rounds()
+    player_1_score = 0
+    player_2_score = 0
+
+    for i in range(rounds_want_to_play):
+        print(f"----- {i+1}. kör -----")                        #VISSZAADJA, HOGY MELYIK KÖR KÖVETKEZIK
+        player_1_choice = get_players_1_choice(player_1)        
+        player_2_choice = get_players_2_choice(player_2)
+
+
+#KIÍRJUK, HOGY MELYIK JÁTÉKOS NYERTE A KÖRT
+        winner = determine_winner(player_1_choice, player_2_choice)
+
+        if winner == 1:
+            print(f"{player_1} won this round!")
+            player_1_score += 1
+        elif winner == 2:
+            print(f"{player_2} won this round!")
+            player_2_score += 1
+        else:

```

Comment:
Logikailag a megoldás szuper, viszont életszerűbb lenne a program, ha nem lehetne döntetlen egy kör, ezáltal a végeredmény sem.
Ehhez csak néhány apró változtatás kell:

```
    current_round = 1

    #for i in range(rounds_want_to_play):
    while current_round <= rounds_want_to_play:
        print(f"----- {current_round}. kör -----")                        #VISSZAADJA, HOGY MELYIK KÖR KÖVETKEZIK
        player_1_choice = get_players_1_choice(player_1)        
        player_2_choice = get_players_2_choice(player_2)


#KIÍRJUK, HOGY MELYIK JÁTÉKOS NYERTE A KÖRT
        winner = determine_winner(player_1_choice, player_2_choice)

        if winner == 1:
            print(f"{player_1} won this round!")
            player_1_score += 1
        elif winner == 2:
            print(f"{player_2} won this round!")
            player_2_score += 1
        else:
            print("This round is a tie, give it another shot!")
            continue
        current_round += 1
```

Ahogy látod annyit változtatnék, hogy van egy current_round változónk. A for helyett while ciklus pörög amíg ez a current_round el nem éri a megadott játszani kívánt kör számot. A trükk ott van, hogy amikor ugyanazt a választ adják, nem csak kiprinteljük hogy "This round is a tie", hanem continue-val gyakorlatilag újrakezdjük az adott iterációt. Azért újrakezdés jelen esetben, mert így nem jut el a kód odáig, hogy növelje a current_round értékét az if-else után. Így biztosítjuk hogy az adott kör mindig újraindul ha ugyanazt adják meg. Ezáltal a végeredmény sem lehet döntetlen, így még rövidebb lesz a kódod (nem kell a 81. soron az else), és a valóságot is jobban tükrözi.